temporalio/lib/temporalio/internal/client/implementation.rb

```diff
@@ -327,26 +327,43 @@ def _start_workflow_request_from_with_start_options(klass, start_options)
         end
 
         def list_workflows(input)
```

Comment:
I think we can probably delete this (and the `ListWorkflowsInput` class) and inline this logic in the client itself. I don't think we need _both_ calls to be interceptable, only the lower level one (the page one)

Reply:
Right, I was debating whether to put that logic in the interceptor or client. I think it makes sense to have in the client, the only downside is this is the first method (that I know of) that is on the client's public interface, but that you can't intercept. Might be a bit confusing, but so is having multiple interceptions caught for a single `list_workflows` call. So I made the change as you suggested!

Reply:
> I think it makes sense to have in the client, the only downside is this is the first method (that I know of) that is on the client's public interface

`execute_workflow` and `execute_update_with_start_workflow` are other examples of just sugar/shortcuts using other client calls